### ISSUE TITLE
Empty item in env field inside c_cpp_properties.json causes error

### ISSUE THREAD

**Issue by schperplata** [Microsoft/vscode#58760]
Issue Type: <b>Bug</b>

I'm playing around with using VS Code for embedded project. I have created a c_cpp_properties.json file, where I am planing to list all directories and files inside of different "env" fields and than passing this values to "configuration" "includePath", "browse" and "define" fields.

The problem specifically is caused by an empty field, followed by a non empty field, see example below:
```
{
    "env": {
        ...
        "userDefines": [],
        ...
        "cubeMXDefines": [
            "USE_HAL_DRIVER",
            "STM32F078xx"
        ],
        ...
    },
    "configurations": [
        {
            ...
            "defines": [
                "${userDefines}",
                "${cubeMXDefines}"
            ],
            ...
        }
    ],
    "version": 4
}
```
As you can see, **userDefines** field is empty, while **cubeMXDefines** in not. While editing main.c for example, I get an error in problems tab:
![defines error](https://user-images.githubusercontent.com/8736111/45594826-5fa01500-b9a1-11e8-92ac-38e4c9818bcd.PNG)

If I write anything into **userDefines** field, Intellisense works as expected. 
Is there something I am missing, or is this a bug (not handled empty env field)?

VS Code version: Code 1.27.2 (f46c4c469d6e6d8c46f268d1553c5dc4b475840f, 2018-09-12T16:17:45.060Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-2520M CPU @ 2.50GHz (4 x 2494)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.95GB (3.16GB free)|
|Process Argv|C:\Users\domen\AppData\Local\Programs\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
bracket-pair-colorizer|Coe|1.0.60
CppSnippets|har|0.0.13
cortex-debug|mar|0.1.21
python|ms-|2018.8.0
cpptools|ms-|0.18.1
debug|web|0.22.0

</details>

**Comment by schperplata** [comment]
@joaomoreno Can you explain more about what extension is causing this issue? 

Just a thought:  I've make an effort to report a possible bug. Please be kind enough and try to write a quick, helpful reply. This vscodebot reply is really not useful at all. 

Thank you.

**Comment by joaomoreno** [comment]
You seem to be writing C code. Did you install an extension to handle C code? Maybe https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools ?

**Comment by joaomoreno** [comment]
@schperplata Absolutely! You can file issues in this repository when it's about the core VS Code product. If it is related to an extension, then the extension's repository is the appropriate place. Thanks for understanding.

**Comment by schperplata** [comment]
OK! I will report it in C/C++ extension then. 
It is quite hard for us users to separate what is VS Code bug and what is extension bug though. 
Anyway, thank you again!
